### DISCUSSION TITLE
Possible to run Argo Workflows in cluster mode

### DISCUSSION THREAD

**Discussion by norman-zon** [argoproj/argo-helm#2991] [Q&A]
Ist it possible, using the helm chart, to run argo-workflows in cluster mode?

As far as I can tell from the chart it is not possible to set `singleNamespace: false` and also unset `workflowNamespaces`. 

My goal is to have the argo-workflow controller watch all namespaces.

**Comment by yu-croco** [comment]
>Ist it possible, using the helm chart, to run argo-workflows in cluster mode?

If you set `.Values.singleNamespace` as `false`, it works as `cluster scoped` .
Ref: https://argo-workflows.readthedocs.io/en/latest/managed-namespace/

>As far as I can tell from the chart it is not possible to set singleNamespace: false and also unset workflowNamespaces.

As [the doc](https://github.com/yu-croco/argo-helm/blob/909ece4d5937c25ee663dc0befad9811ef6e9ebe/charts/argo-workflows/values.yaml#L273-L277) says, you are supposed to set all namespaces in `workflowNamespaces` instead of unset, or you can create SA in individual namespaces.
Ref: https://github.com/argoproj/argo-helm/tree/main/charts/argo-workflows#serviceaccount-for-workflow-spec

**Reply by norman-zon** [reply]
Thanks for the explanation.
I am not so much worried about having to create the SAs manually, but rather the according `Role` and `RoleBinding`. But as far as I understand there is no way of using a `ClusterRole` and `ClusterRoleBinding` instead of separate ones per namespace?

**Reply by yu-croco** [reply]
>But as far as I understand there is no way of using a ClusterRole and ClusterRoleBinding instead of separate ones per namespace?

Yes, you are right.
argo-helm follows upstream's manifest and upstream provides Role/RoleBinding instead of cluster scope, so I think it's inhered as Role/RoleBinding in argo-helm. 🤔 
https://github.com/argoproj/argo-workflows/blob/eb23eb6b84a1ecf0826a4e16d6a2032fd0cb08b7/manifests/quick-start-minimal.yaml#L2735-L2757
